web_editor/js/main.js

```diff
@@ -0,0 +1,147 @@
+// TODO The hash changes, this is very brittle. See
+// https://github.com/thedodd/trunk/issues/230 or stop using trunk.
+import { dummyData } from "./dummy_data.js";
+import init, {
+  js_way_to_lanes,
+  js_lanes_to_tags,
+} from "./osm2lanes-npm/osm2lanes_npm.js";
+await init();
+
+// Setup the toolbox controls
+new Sortable(document.getElementById("toolbox"), {
+  group: {
+    name: "toolbox",
+    pull: "clone",
+  },
+  sort: false,
+  animation: 150,
+  ghostClass: "card-being-dragged",
+});
+
+// TODO Disable the ghostClass of lanes when going here
+new Sortable(document.getElementById("delete"), {
+  group: "lanes",
+  onAdd: function (evt) {
+    var el = evt.item;
+    el.parentNode.removeChild(el);
+  },
+});
+
+// This state is set up by start_editing
+var current_road = null;
+var current_locale = null;
+
+async function start_editing() {
+  const way = BigInt(document.getElementById("osm_way_id").value);
+  // Faster dev workflow: if the way ID is the default, use baked-in data instead of waiting on Overpass.
+  var road_wrapper, locale;
+  if (way == 804788513) {
+    [road_wrapper, locale] = dummyData();
+  } else {
+    console.log(`Fetching ${way}...`);
+    [road_wrapper, locale] = await js_way_to_lanes(way);
+  }
+  const road = road_wrapper["Ok"]["road"];
+  console.log(`Got osm2lanes output, creating cards`);
+
+  current_road = JSON.parse(JSON.stringify(road));
+  current_locale = JSON.parse(JSON.stringify(locale));
+
+  // TODO Fully clean up the old cards, including whatever sortable thing is attached there?
+  const cards = document.getElementById("cards");
+  cards.replaceChildren();
+
+  // Create a card per lane
+  for (const lane of road["lanes"]) {
+    if (lane["type"] == "separator") {
+      continue;
+    }
+    cards.appendChild(make_lane_card(lane));
+  }
+
+  new Sortable(cards, {
+    group: {
+      name: "lanes",
+      put: ["toolbox"],
+    },
+    animation: 150,
+    ghostClass: "card-being-dragged",
+    onAdd: function (evt) {
+      const type = evt.item.getAttribute("value");
+      // TODO switch case but without break?
+      // TODO Figure out direction based on center line position
+      var card;
+      if (type == "driving") {
+        card = make_lane_card({
+          type: "travel",
+          direction: "backward",
+          designated: "motor_vehicle",
+        });
+      } else if (type == "bicycle") {
+        card = make_lane_card({
+          type: "travel",
+          direction: "backward",
+          designated: "bicycle",
+        });
+      } else if (type == "parking") {
+        card = make_lane_card({
+          type: "parking",
+          direction: "backward",
+          designated: "motor_vehicle",
+        });
+      }
+      evt.item.replaceWith(card);
+    },
+  });
+}
+window.start_editing = start_editing;
+
+function make_lane_card(lane) {
```

Comment:
Zero attempt at style yet, obviously

Reply:
how about code style? 😬 

```suggestion
function makeLaneCard(lane) {
```

It's not unheard of for people to write javascript like it's c, but it's not typical. I suppose it ultimately depends if you want this web code to feel more familiar to rust programmers or to web programmers.

Reply:
Oops, fixed. I have to un-Rust my brain, or find a linter that yells at me.